airflow/configuration.py

```diff
@@ -369,59 +381,86 @@ def as_dict(
             command to run (False)
         :type include_cmds: bool
         """
-        cfg = {}
+        config = {}
         configs = [
             ('default', self.airflow_defaults),
             ('airflow.cfg', self),
         ]
 
         for (source_name, config) in configs:
             for section in config.sections():
-                sect = cfg.setdefault(section, OrderedDict())
-                for (k, val) in config.items(section=section, raw=raw):
-                    if display_source:
-                        val = (val, source_name)
-                    sect[k] = val
+                self._set_config_section(config, display_source, raw, section, source_name)
```

Comment:
Why?

Reply:
Was trying to reduce the depth (pylint error) by extracting the logic into separate functions